pkg/core/chain/abci.go

```diff
@@ -160,6 +161,17 @@ func (app *CoreApplication) FinalizeBlock(ctx context.Context, req *abcitypes.Fi
 				app.logger.Errorf("failed to persist tx stat: %v", err)
 			}
 
+			if vr := signedTx.GetValidatorRegistration(); vr != nil {
+				validatorUpdates = append(
+					validatorUpdates,
+					abcitypes.ValidatorUpdate{
+						Power:       vr.Power,
+						PubKeyBytes: vr.PubKey,
+						PubKeyType:  "ed25519",
```

Comment:
nit: this magic string is probably fine, there's no const we can import from cometbft or anything right?